src/pybind/mgr/cephadm/services/osd.py

```diff
@@ -110,11 +119,17 @@ def deploy_osd_daemons_for_existing_osds(self, host: str, service_name: str,
         fsid = self.mgr._cluster_fsid
         osd_uuid_map = self.mgr.get_osd_uuid_map()
         created = []
+        size = 0
         for osd_id, osds in osds_elems.items():
             for osd in osds:
                 if osd['tags']['ceph.cluster_fsid'] != fsid:
                     logger.debug('mismatched fsid, skipping %s' % osd)
                     continue
+                osdspec_affinity = osd['tags'].get('ceph.osdspec_affinity', "")
+                if (osdspec_affinity == service_name[4:]) or \
+                   ((osdspec_affinity == "" or osdspec_affinity is None) and (service_name[4:] == 'unmanaged' or service_name[4:] == '')):
```

Comment:
This loop iterates over the list of devices or logical volumes associated with osds in each host( "ceph-volume lvm list" command output). We can have osds created by different "osd.services" , so we only must count those "items"  that belongs to the service passed as parameter in "deploy_osd_daemons_for_existing_osds".
 
Only one exception...  osd devices/lvs that are not directly associated to an "osd service" (osdspec_affinity is empty or "unmanaged", we only must count these items if the "service_name" parameter is empty or "unmanaged".

Reply:
ah, okay. I was confused by the `deploy_osd_daemons_for_existing_osds` function and thought it was just looping over new osds to be created since it calls `_create_daemon` later on but I see now it's looping through all the osds on the host and just dropping out at the `if osd_id in before_osd_uuid_map and osd_id not in replace_osd_ids` check if the osd is already created and up so `_create_daemon` is not called again. In that case it definitely makes sense to be checking the service name if we're going to use the `size` variable here to eventually update the size of the osd service.